### PR TITLE
Implement private school VAT

### LINKED ISSUE #879: Add variable and parameter to simulate taxing private school fees

### PR DESCRIPTION
**Pull request by anth-volk**
Fixes #879.

This is a first take at implementing the private school VAT proposal from the Labour Party's 2024 manifesto.

### CODE REVIEW

**Comment by anth-volk** [comment]
This does not yet implement private school attendance rates sourced from external reports or the actual VAT rate

File: policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+description: Mean annual private school fee
+values:
+  2022-01-01: 15_200
+metadata:
+  unit: currency-GBP
```
Comment (nikhilwoodruff) [inline policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml]: We'll want to add a projection onto future years here.

File: policyengine_uk/variables/contrib/labour/private_school_vat.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from policyengine_uk.model_api import *
+from sys import stderr
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/contrib/labour/private_school_vat.py]: Unintentional?

File: policyengine_uk/variables/contrib/labour/private_school_vat.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class private_school_vat(Variable):
+    label = "Private school VAT"
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/contrib/labour/private_school_vat.py]: `label` should start lower-case (you should be able to drop `label` mid-sentence)

**Comment by nikhilwoodruff** [comment]
image

https://www.tandfonline.com/doi/full/10.1080/09645292.2021.1874878

File: policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml
Comment (nikhilwoodruff) [inline policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml:1]: Also let's add a parameter for total private school kids

File: policyengine_uk/variables/contrib/labour/private_school_vat.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                household_income_decile
+            ]
+        )
+        avg_yearly_private_school_cost = parameters(
+            period
+        ).calibration.programs.private_school_vat.private_school_fees
+        # num_children =  How do we count children? num_children is a benunit var
+
+        # The below is incorrect
+        num_people: int = household("household_count_people", period)
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/contrib/labour/private_school_vat.py]: For num kids:

```python
add(household, period, ["is_child"])
```
or even
```python
household.sum(household.members("is_child", period))
```

File: policyengine_uk/variables/contrib/labour/private_school_vat.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            "household_income_decile", period
+        )
+        attends_private_school: bool = (
+            random(household)
+            < parameters(
+                period
+            ).calibration.programs.private_school_vat.private_school_attendance_rate[
+                household_income_decile
+            ]
+        )
```
Comment (nikhilwoodruff) [inline policyengine_uk/variables/contrib/labour/private_school_vat.py]: CODE IN

percent_in_private_school_by_percentile = [0, 0, 0, ..., 0.45, 0.5]
ADJUSTMENT_FACTOR = 1
adjusted_percent_in_private_school_by_percentile *= ADJUSTMENT_FACTOR

# calculate household income percentile
# map to percent in private school
# put kids in private school according to that percentage (mostly done here except percentile not decile)
...
TESTING_CODE
is_child = [False, False, True, False, ...]
child_household_weight = [500, 1000, 2000, 700, ...]
is_in_private_school = [False, False, True, False, ...]
(is_in_private_school * child_household_weight).sum()

model_predicted_kids_in_private_school = 800k # should be 600k

File: policyengine_uk/variables/contrib/labour/private_school_vat.py (hunk 2 above)
Comment (anth-volk) [reply policyengine_uk/variables/contrib/labour/private_school_vat.py]: That it is haha

File: policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml (hunk 1 above)
Comment (anth-volk) [reply policyengine_uk/parameters/calibration/programs/private_school_vat/private_school_fees.yaml]: Could you describe that a bit more? The source of this number only gave a number for the 2022/23 year, hence it being put in like that

**Comment by anth-volk** [comment]
Note: To avoid a cycle in the tax/benefit logic, the most recent iteration uses `household_market_income` to calculate likelihood to enroll in private school, not `household_net_income`. Technically, this does not accord 100% with the data used to estimate enrollment likelihood.

**Review by nikhilwoodruff — APPROVED** [review]
Amazing- added a few modifications but looks good

**Comment by nikhilwoodruff** [comment]
Tests passed

**Pull request merged**